packages/backend-plugin-api/src/services/definitions/AuditorService.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ErrorLike } from '@backstage/errors';
+import type { JsonObject } from '@backstage/types';
+import type { Request } from 'express';
+
+/** @public */
+export type AuditorEventStatus<E = ErrorLike> =
+  | { status: 'unknown' }
+  | { status: 'initiated' }
+  | { status: 'succeeded' }
+  | ({
+      status: 'failed';
+    } & ({ error: E } | { errors: E[] }));
+
+/**
+ * Options for creating an auditor event.
+ *
+ * @public
+ */
+export type AuditorEventOptions<T extends JsonObject> = {
+  /**
+   * Use kebab-case to name audit events (e.g., "user-login", "file-download").
+   *
+   * The `pluginId` already provides plugin/module context, so avoid redundant prefixes in the `eventId`.
+   */
+  eventId: string;
+
+  /** (Optional) The associated HTTP request, if applicable. */
+  request?: Request<any, any, any, any, any>;
+
+  /** (Optional) An identifier for the entity or user who triggered the event. */
+  actorId?: string;
+
+  /** (Optional) Additional metadata relevant to the event, structured as a JSON object. */
+  meta?: T;
+} & AuditorEventStatus;
+
+/** @public */
+export type AuditorEventLevels = keyof Omit<
+  AuditorService,
+  'createAuditorEvent'
+>;
+
+/** @public */
+export type AuditorCreateEvent<
+  T extends JsonObject,
+  E = ErrorLike,
+> = (options: {
+  /**
+   * Use kebab-case to name audit events (e.g., "user-login", "file-download").
+   *
+   * The `pluginId` already provides plugin/module context, so avoid redundant prefixes in the `eventId`.
+   */
+  eventId: string;
+
+  level?: Exclude<AuditorEventLevels, 'error'>;
```

Comment:
I don't feel that this lines up with [the discussion around the levels](https://github.com/backstage/backstage/pull/26372#discussion_r1785239022). This implies that the `error` level is only used for reporting errors, but just because something resulted in an error doesn't mean it's important from a security PoV.

I'm feeling that it would be more useful to replace this with a "security sensitivity" scale, where most events probably fall under a default level, but certain endpoints, in particular important write endpoints, would be able to assign a higher level. Honestly that's kind of baked into the event ID though, but might be useful to have as directly accessible metadata too.

I had a look at some examples from GCP: https://cloud.google.com/logging/docs/audit/gsuite-login-samples#samples. They use the regular log infra for audit logs too, so it's a pretty good comparison. In their case all audit logs are of the `NOTICE` level, regardless of the security severity of the event.

Reply:
@Rugvip What do you think about the current implementation?